Unspecified float/int append X

ax3l opened the pull request:
This is a wording change of the openPMD 1.0.0 standard as proposed in #146.

## Description

When we are referring to floating point numbers with unspecified (user-chosen) precision, we use `(floatX)` for `(float16|32|64|128|...)` now to avoid confusion with single precision "C" floats. The same change is added for the type of (unsigned) integers.

Also replaces the usage of C/C++/Python `double` with `float64` for consistency.

## Affected Components

- `base`
- extension: `ED-PIC`

## Logic Changes

No changes introduced.

## Writer Changes

No data writers affected.

## Reader Changes

No data readers affected.

## Data Converter

No data conversion needed.

ax3l committed:
Unspecified float/int append X

This is a rewording change of the openPMD 1.0.0 standard.

Description

When we are referring to floating point numbers with unspecified
(user-chosen) precision, we use `(floatX)` for
`(float16|32|64|128|...)` now to avoid confusion with single
precision "C" floats.

Logic Changes

No changes introduced.

Writer Changes

No data writers affected.

Reader Changes

No data readers affected.

ax3l commented:
@RemiLehe ping :) please review & merge me :)

RemiLehe commented on STANDARD.md:
```diff
@@ -12,9 +12,9 @@ interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
 All `keywords` in this standard are case-sensitive.
 
-The naming *(float)* without a closer specification is used if the implementor
+The naming *(floatX)* without a closer specification is used if the implementor
```
I think the word *closer* is ambiguous here. I guess you mean `without any further specification`. But this applied to `float` (which did not have any further specification, like `float16`). With the new notation `floatX`, I am not sure the phrase `without a closer specification` makes sense anymore, it the sense that `X` is, in a way, a specification.

ax3l replied:
That's a great point, we could maybe even shorten it to just `without further specification`?

RemiLehe commented on STANDARD.md:
```diff
@@ -12,9 +12,9 @@ interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
 
-The naming *(float)* without a closer specification is used if the implementor
+The naming *(floatX)* without a closer specification is used if the implementor
 can choose which kind of floating point precision shall be used.
```
Here, just for clarification, I would add examples:
```
can choose which kind of floating point precision shall be used (e.g. float32, float64, float128, etc.)
```

ax3l replied:
Good point, makes it less abstract.

ax3l replied:
It would probably make sense to then rename all the `double` namings to `float64` explicitly as we already do it for the Fortran equivalents.
C++ is not to sane in defining it's "double" definition and we really do mean IEEE-754 float64 anyway where used.

RemiLehe commented on STANDARD.md:
```diff
@@ -12,9 +12,9 @@ interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+The naming *(floatX)* without a closer specification is used if the implementor
 can choose which kind of floating point precision shall be used.
-The naming *(uint)* and *(int)* without a closer specification is used if the
+The naming *(uintX)* and *(intX)* without a closer specification is used if the
```
The above comments for `float` also apply for `int` and `uint`

RemiLehe commented:
Thanks for the clarifications in this pull request. I have a few minor requests (see inline comments)

ax3l committed:
floatX: review comments

and adds examples.
including the same for uintX and intX

ax3l committed:
double -> float64

rewrites the usage for double with float64 for consistency.

ax3l commented:
Thank you for the comments, I adjusted the wording accordingly to remove ambiguity and added the examples.

For consistency, I also renamed the few usages of `double` for double precision with `float64`.

RemiLehe commented:
Thanks for updating the PR! I'll merge now.
